src/lib/structures/expr.ml

```diff
@@ -2682,9 +2683,6 @@ module Purification = struct
           | _ -> failwith "unexpected expression in purify_form"
         end
 
-      | Sy.Op Sy.Optimize _ ->
-        purify_literal e
```

Comment:
Is this purification performed somewhere else?

Reply:
oups I forget this one. I'll check it now :)